openapi2beans/pkg/generator/schema2package.go

```diff
@@ -0,0 +1,200 @@
+	}
+	return enumValues
+}
+
+func retrieveDataMembersFromSchemaType(schemaType *SchemaType) (dataMembers []*DataMember, requiredMembers []*RequiredMember, constantDataMembers []*DataMember){
+	for _, property := range schemaType.properties {
+		var constVal string
+		name := property.name
+		description := strings.Split(property.description, "\n")
+		if len(description) == 1 && description[0] == "" {
+			description = nil
+		} else if len(description) > 1 {
+			description = description[:len(description)-2]
+		}
+		if property.IsConstant() {
+			posVal := possibleValuesToEnumValues(property.GetPossibleValues())
+			name = convertToConstName(name)
+			constVal = convertConstValueToJavaReadable(posVal[0], property.typeName)
+
+			constDataMember := DataMember {
+				Name: name,
+				CamelCaseName: convertToCamelCase(name),
+				MemberType: propertyToJavaType(property),
+				Description: description,
+				ConstantVal: constVal,
+			}
+
+			constantDataMembers = append(constantDataMembers, &constDataMember)
+
+		} else {
+
+			dataMember := DataMember {
+				Name: name,
+				CamelCaseName: convertToCamelCase(name),
+				MemberType: propertyToJavaType(property),
+				Description: description,
+				ConstantVal: constVal,
+			}
+			dataMembers = append(dataMembers, &dataMember)
+				
+			if property.IsSetInConstructor() {
+				requiredMember := RequiredMember {
+					DataMember: &dataMember,
+				}
+				requiredMembers = append(requiredMembers, &requiredMember)
+			}
+		}
+		
+	}
+	sort.SliceStable(dataMembers, func (i int, j int) bool {return isDataMemberLessThanComparison(dataMembers[i], dataMembers[j])})
+	sort.SliceStable(requiredMembers, func (i int, j int) bool {return isDataMemberLessThanComparison(requiredMembers[i].DataMember, requiredMembers[j].DataMember)})
+	sort.SliceStable(constantDataMembers, func (i int, j int) bool {return isDataMemberLessThanComparison(constantDataMembers[i], constantDataMembers[j])})
+	if requiredMembers != nil {
+		requiredMembers[0].IsFirst = true
+	}
+	return dataMembers, requiredMembers, constantDataMembers
+}
+
+func propertyToJavaType(property *Property) string {
+	javaType := ""
+	if property.IsReferencing() || property.typeName == "object" || property.IsEnum() {
+		javaType = property.resolvedType.name
+	} else {
+		if property.typeName == "string" {
+			javaType = "String"
+		} else if property.typeName == "integer" {
+			javaType = "int"
+		} else if property.typeName == "number" {
+			javaType = "double"
+		}else if property.typeName == "" {
+			javaType = "Object"
+		} else {
+			javaType = property.typeName
+		}
+	}
+
+	if property.IsCollection() {
+		dimensions := property.cardinality.max / 128
+		for range dimensions {
+			javaType += "[]"
+		}
+	}
+
+	return javaType
+}
+
+func convertToCamelCase(name string) string {
+	initialLetter := name[0]
+	camelCaseName := strings.ToUpper(string(initialLetter)) + name[1:]
+	return camelCaseName
+}
+
+func convertToConstName(name string) string {
```

Comment:
It's not easy to see what this is doing. Can you comment with some example inputs/outputs pls ?